alf/algorithms/lagrangian_reward_weight_algorithm.py

```diff
@@ -103,7 +102,7 @@ def _inv_softplus(self, tensor):
     def reward_weights(self):
         """Return the detached reward weights. These weights are expected not to
         be changed by external code."""
-        return self._reward_weights.detach().clone()
+        return F.softplus(self._lambdas).detach().clone()
```

Comment:
This function can be called after class init so if self._lambdas is updated by checkpoint, we should always return the newest value. For simplicity, we don't want a variable to cache its latest value.